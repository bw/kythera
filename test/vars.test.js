require("jest")
const common = require("./common")

let i = 0

afterEach(() => {
	i += 1
})

describe("Variables", () => {
	let program = common.parseFile("test/vars.ky")
	i = 0

	let t = (name) => {
		test(name, () => expect(program[i]).toMatchSnapshot())
	}

	describe("Boolean primitive", () => {
		t("declaration")
		t("assignment")
	})

	describe("Integer primitive", () => {
		t("declaration")
		t("assignment")
	})

	describe("Floating point primitive", () => {
		t("declaration")
		t("assignment")
	})

	describe("String", () => {
		t("declaration")
		t("assignment")
	})

	describe("null", () => {
		t("declaration")
		t("read")
	})

	describe("Function", () => {
		t("declaration")
	})

	describe("using 'new'", () => {
		t("int")
		t("float")
		t("bool")
		t("str")
	})
})