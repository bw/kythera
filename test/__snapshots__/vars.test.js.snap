// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Variables Boolean primitive assignment 1`] = `
Object {
  "kind": "assign",
  "left": Object {
    "kind": "identifier",
    "name": "myBool",
  },
  "operator": "=",
  "right": Object {
    "kind": "literal",
    "type": "bool",
    "value": true,
  },
}
`;

exports[`Variables Boolean primitive declaration 1`] = `
Object {
  "identifier": "myBool",
  "kind": "let",
  "value": Object {
    "kind": "literal",
    "type": "bool",
    "value": false,
  },
}
`;

exports[`Variables Floating point primitive assignment 1`] = `
Object {
  "kind": "assign",
  "left": Object {
    "kind": "identifier",
    "name": "myFloat",
  },
  "operator": "=",
  "right": Object {
    "kind": "literal",
    "type": "float",
    "value": 2.71828183,
  },
}
`;

exports[`Variables Floating point primitive declaration 1`] = `
Object {
  "identifier": "myFloat",
  "kind": "let",
  "value": Object {
    "kind": "literal",
    "type": "float",
    "value": 3.14159,
  },
}
`;

exports[`Variables Function declaration 1`] = `
Object {
  "identifier": "myFn",
  "kind": "let",
  "value": Object {
    "body": Array [
      Object {
        "kind": "return",
        "value": Object {
          "kind": "literal",
          "type": "str",
          "value": "abc",
        },
      },
    ],
    "kind": "literal",
    "parameters": Array [
      Object {
        "name": "a",
        "type": Object {
          "kind": "type",
          "origin": "builtin",
          "type": "int",
        },
      },
    ],
    "returns": Object {
      "kind": "type",
      "origin": "builtin",
      "type": "str",
    },
    "type": "fn",
  },
}
`;

exports[`Variables Integer primitive assignment 1`] = `
Object {
  "kind": "assign",
  "left": Object {
    "kind": "identifier",
    "name": "myInt",
  },
  "operator": "=",
  "right": Object {
    "kind": "literal",
    "type": "int",
    "value": 3,
  },
}
`;

exports[`Variables Integer primitive declaration 1`] = `
Object {
  "identifier": "myInt",
  "kind": "let",
  "value": Object {
    "kind": "literal",
    "type": "int",
    "value": 99,
  },
}
`;

exports[`Variables String assignment 1`] = `
Object {
  "kind": "assign",
  "left": Object {
    "kind": "identifier",
    "name": "myStr",
  },
  "operator": "=",
  "right": Object {
    "kind": "literal",
    "type": "str",
    "value": "world",
  },
}
`;

exports[`Variables String declaration 1`] = `
Object {
  "identifier": "myStr",
  "kind": "let",
  "value": Object {
    "kind": "literal",
    "type": "str",
    "value": "hello",
  },
}
`;

exports[`Variables null declaration 1`] = `
Object {
  "identifier": "myNull",
  "kind": "let",
  "value": Object {
    "kind": "literal",
    "type": "null",
    "value": null,
  },
}
`;

exports[`Variables null read 1`] = `
Object {
  "kind": "identifier",
  "name": "myNull",
}
`;

exports[`Variables using 'new' bool 1`] = `
Object {
  "identifier": "myNewBool",
  "kind": "let",
  "value": Object {
    "kind": "new",
    "target": Object {
      "kind": "type",
      "origin": "builtin",
      "type": "bool",
    },
  },
}
`;

exports[`Variables using 'new' float 1`] = `
Object {
  "identifier": "myNewFloat",
  "kind": "let",
  "value": Object {
    "kind": "new",
    "target": Object {
      "kind": "type",
      "origin": "builtin",
      "type": "float",
    },
  },
}
`;

exports[`Variables using 'new' int 1`] = `
Object {
  "identifier": "myNewInt",
  "kind": "let",
  "value": Object {
    "kind": "new",
    "target": Object {
      "kind": "type",
      "origin": "builtin",
      "type": "int",
    },
  },
}
`;

exports[`Variables using 'new' str 1`] = `
Object {
  "identifier": "myNewStr",
  "kind": "let",
  "value": Object {
    "kind": "new",
    "target": Object {
      "kind": "type",
      "origin": "builtin",
      "type": "str",
    },
  },
}
`;
